mapillary.py

```diff
@@ -44,15 +58,37 @@ def run(bbox: str, traces_source: any, processes: int):
         # TODO: Delete the tmp dir after run?
 
 
-def to_bbox(llo, lla, mlo, mla):
+def to_bbox(llo: float, lla: float, mlo: float, mla: float) -> str:
+    """
+    Given (min_lon, min_lat, max_lon, max_lat) bounding box values, returns a string representation understood by
+    Mapillary APIs.
+    """
     return ','.join([str(llo), str(lla), str(mlo), str(mla)])
 
 
+def is_within_bbox(lon: float, lat: float, bbox: list[float]) -> bool:
+    """
+    Checks if lon / lat coordinate is within a bbox in the format of [min_lon, min_lat, max_lon, max_lat]
+    """
+    return bbox[0] <= lon < bbox[2] and bbox[1] <= lat < bbox[3]
+
+
 def initialize_multiprocess(global_output_dir_: str, global_output_tmp_dir_: str, global_traces_source_: any,
-                            finished_bbox_sections_: multiprocessing.Value):
-    # For persistent connections
+                            finished_bbox_sections_: multiprocessing.Value) -> None:
+    """
+    Initializes global variables referenced / updated by all threads of the multiprocess API requests.
+    """
+    # For persistent connections and timeout settings
     global session
     session = requests.Session()
+    retry_strategy = Retry(
+        total=5,
+        status_forcelist=[429, 500, 502, 503, 504],
+        backoff_factor=3
+    )
+    adapter = HTTPAdapter(max_retries=retry_strategy)
+    session.mount('https://', adapter)
+    session.mount('http://', adapter)
```

Comment:
Retry with backoff as a default on all API calls

Reply:
cool i didnt know there was a non manual way to do this